pepsico/monthly_seasonal_conv_mapping.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import plotly.express as px
+from dash import Dash, dcc, html, Input, Output
+from urllib.request import urlopen
+import json
+
+def compute_annual_monthly_avg(ds, variable, start_year=None, end_year=None):
+    #compute the avg (variable) for each month across the specified years
+    
+    #if precipitation variable, change from kg to mm per day
+    if variable == 'pr':
+        ds[variable] *= 86400
+    
+    if start_year and end_year:
+        ds = ds.sel(T=slice(f"{start_year}-01-01", f"{end_year}-12-31"))
```

Comment:
sel's slice accepts None (does nothing) and will select the full year if you give the year only as label, that is to say if input is 2010, will select all of 2010. In other words, you need not the "-01-01" and the "-12-31" and because None is an acceptable input, you need not the if statement and can just apply

ds = ds.sel(T=slice(f"{start_year}", f"{end_year}"))